### PR TITLE
Add optional context to remoteValidationError

### PR DESCRIPTION
**Pull request by JamyDev**

### CODE REVIEW

File: src/adapters/errors/RemoteValidationError.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ export interface IValidationErrorDetail {
     path: string;
     type: string; // tslint:disable-line
     message: string;
+    context?: {
+        [key: string]: any
```
Comment (SimonSchick) [inline src/adapters/errors/RemoteValidationError.ts]: Please narrow, this can only be a JSON type.

Comment (JamyDev) [inline src/adapters/errors/RemoteValidationError.ts]: Do we have a type for that?

File: src/Types.ts [hunk 2]
```diff
@@ -599,3 +599,8 @@ export interface IEmbedOptions<T extends IModel> {
     multi?: boolean;
     model: ForwardRef<IModelCtor<T>>;
 }
+
+export type JSONPrimitive = boolean | string | number | null;
+export type IJSON = {
```
Comment (UnwrittenFun) [inline src/Types.ts]: This should be an interface

Comment (JamyDev) [inline src/Types.ts]: tru

Comment (SimonSchick) [inline src/Types.ts]: My bad actually 😛

**Pull request merged**